virtcontainers/kata_agent.go

```diff
@@ -252,6 +253,11 @@ func (k *kataAgent) configure(h hypervisor, id, sharePath string, builtin bool,
 		k.proxyBuiltIn = true
 	}
 
+	caps := h.capabilities()
+	if !caps.is9pSupported() {
+		return nil
+	}
+
 	// Adding the shared volume.
 	// This volume contains all bind mounted container bundles.
```

Comment:
Please extend the comment here, saying that this code path should only be executed when 9p is supported.